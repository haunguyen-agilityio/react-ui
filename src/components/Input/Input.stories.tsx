//Libraries
import { ComponentStory, ComponentMeta } from '@storybook/react'

// Components
import Input from '../Input'

export default {
  title: 'Components/Input',
  component: Input,
} as ComponentMeta<typeof Input>

const Template: ComponentStory<typeof Input> = (args) => <Input {...args} />

export const Primary = Template.bind({})
Primary.args = {
  label: 'Default',
  placeholder: 'Placeholder',
}

export const Disabled = Template.bind({})
Disabled.args = {
  label: 'Disabled',
  placeholder: 'Placeholder',
  isDisable: true,
}

export const Filled = Template.bind({})
Filled.args = {
  value: 'Placeholder',
  label: 'Filled',
  placeholder: 'Placeholder',
}

export const Error = Template.bind({})
Error.args = {
  value: 'Placeholder',
  label: 'Error',
  placeholder: 'Placeholder',
  isError: true,
  errorText: 'Required',
}
