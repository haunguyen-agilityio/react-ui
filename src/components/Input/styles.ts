// Libraries
import { styled } from '@mui/system'
import TextField from '@mui/material/TextField'
import FormHelperText from '@mui/material/FormHelperText'

// Themes
import theme from '@themes'
import typography from '@themes/typography'

export const InputWrapper = styled('div')({
  padding: theme.spacing(2),
})

export const StyledFormHelperText = styled(FormHelperText)({
  color: theme.palette.error.main,
  fontSize: typography.body1.fontSize,
})

export const StyledTextField = styled(TextField)((props) => ({
  '& .MuiInputBase-input': {
    color: props.error ? theme.palette.error.main : theme.palette.text.primary,
  },
  ' .MuiOutlinedInput-root': {
    '& fieldset': {
      borderRadius: 0,
      padding: theme.spacing(0),
      borderColor: props.error ? theme.palette.error.main : theme.palette.grey[400],
      '&.MuiOutlinedInput-notchedOutline': {
        borderColor: props.error ? theme.palette.error.main : theme.palette.grey[400],
      },
    },
    '&:hover fieldset': {
      border: props.error
        ? `1px solid ${theme.palette.error.main}`
        : `1px solid ${theme.palette.grey[400]}`,
    },
    '&.Mui-focused fieldset': {
      borderColor: theme.palette.grey[400],
      border: props.error
        ? `1px solid ${theme.palette.error.main}`
        : `1px solid ${theme.palette.grey[400]}`,
    },
  },
}))
