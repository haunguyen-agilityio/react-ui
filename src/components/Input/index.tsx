// Libraries
import Typography from '@mui/material/Typography'

// Styles
import { StyledTextField, InputWrapper } from './styles'
import { StyledFormHelperText } from './styles'

// Themes
import theme from '../../themes'

enum TypeCategory {
  Number = 'number',
  Text = 'text',
  Password = 'password',
  Search = 'search',
}

export interface IInputProps {
  value: string
  label: string
  placeholder: string
  isDisable?: boolean
  isError?: boolean
  errorText?: string
  textFieldTypes?: TypeCategory
  onChange: () => void
}

const Input = ({
  value = '',
  label = '',
  placeholder = '',
  isError = false,
  isDisable = false,
  errorText = '',
  textFieldTypes = TypeCategory.Text,
  onChange = () => {},
}: IInputProps) => {
  return (
    <InputWrapper>
      <Typography
        data-testid='title'
        variant='subtitle2'
        color={isError ? theme.palette.error.main : isDisable ? `${theme.palette.grey[400]}` : ''}
      >
        {label}
      </Typography>
      <StyledTextField
        data-testid='input'
        fullWidth
        type={textFieldTypes}
        error={isError}
        disabled={isDisable}
        placeholder={placeholder}
        onChange={onChange}
        value={value}
      />
      <StyledFormHelperText data-testid='errorMessage'>{errorText}</StyledFormHelperText>
    </InputWrapper>
  )
}
export default Input
