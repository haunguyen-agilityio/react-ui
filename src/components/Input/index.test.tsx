// Libraries
import { screen, render } from '@testing-library/react'
import * as renderer from 'react-test-renderer'

// Components
import Input from '@components/Input/index'

describe('Input', () => {
  const defaultProps = {
    value: '',
    label: 'Number',
    placeholder: 'Please input number',
    onChange: () => {},
  }
  it('Component [Input] matches DOM Snapshot', () => {
    const component = renderer.create(<Input {...defaultProps} />)

    expect(component).toMatchSnapshot()
  })

  it('Component [Input] should render correctly', () => {
    render(<Input {...defaultProps} />)

    expect(screen.getByTestId('title')).toHaveTextContent('Number')
    expect(screen.getByPlaceholderText('Please input number')).toBeTruthy()
  })
})
