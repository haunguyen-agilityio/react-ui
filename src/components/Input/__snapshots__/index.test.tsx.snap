// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Component [Input] matches DOM Snapshot 1`] = `
<div
  className="css-1ruxp1v"
>
  <h6
    className="MuiTypography-root MuiTypography-subtitle2 css-1qn9uqj-MuiTypography-root"
    data-testid="title"
  >
    Number
  </h6>
  <div
    className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-1tqk1xf-MuiFormControl-root-MuiTextField-root"
    data-testid="input"
  >
    <div
      className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        autoFocus={false}
        className="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
        disabled={false}
        id=":r0:"
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        placeholder="Please input number"
        required={false}
        type="text"
        value=""
      />
      <fieldset
        aria-hidden={true}
        className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
      >
        <legend
          className="css-hdw1oc"
        >
          <span
            className="notranslate"
          >
            â€‹
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
  <p
    className="MuiFormHelperText-root css-123b7mg-MuiFormHelperText-root"
    data-testid="errorMessage"
  />
</div>
`;
